### PR TITLE
Script changes mostly

### PR DESCRIPTION
**Pull request by heze8**

### CODE REVIEW

File: Assets/Prefabs/Player/Player.prefab [hunk 1]
```diff
@@ -1,4958 +1,5 @@
 %YAML 1.1
 %TAG !u! tag:unity3d.com,2011:
---- !u!1 &1003433082
```
Comment (Eclmist) [inline Assets/Prefabs/Player/Player.prefab]: Player prefab is volatile, what change is this? What has been deleted from the Player prefab?

File: Assets/Scripts/AI/AiPlayer.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+using UnityEngine;
+
+public class AiPlayer : AiActor
+{
+    //public ThirdPersonCharacter m_ThirdPersonCharacter;
+    public void Update()
+    {
+        //m_ThirdPersonCharacter.Move(m_Agent.desiredVelocity, false, false);
+    }
+}
```
Comment (Eclmist) [inline Assets/Scripts/AI/AiPlayer.cs]: A script that has been completely commented out should not be pushed to the repo. Unless this is code removal, then perhaps remove the entire script.

File: Assets/Scripts/Gameplay/GameManager.cs [hunk 3]
```diff
@@ -1,35 +1,41 @@
...
+            yield break;
+        }
+        //Handle spawning items here
+        GameObject item = m_SpawnedItems[Random.Range(0, m_SpawnedItems.Count)];
+        Vector3 spawnPos = m_TerrainWhereItemsSpawn[Random.Range(0, m_TerrainWhereItemsSpawn.Count)].transform.position;
+        //itemsToBeSpawned
+        Instantiate(item, spawnPos, item.transform.rotation);
+        yield return new WaitForSeconds(itemSpawnDelay);
+        StartCoroutine(SpawnItems());
+    }
```
Comment (Eclmist) [inline Assets/Scripts/Gameplay/GameManager.cs]: I feel like spawners should be their own script, and placed individually to handle their own spawn rates etc.

File: Assets/Scripts/AI/States/AiState.cs [hunk 4]
```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+using Random = UnityEngine.Random;
```
Comment (Eclmist) [inline Assets/Scripts/AI/States/AiState.cs]: Remove pls

File: Assets/Scripts/Gameplay/GameManager.cs [hunk 5]
```diff
@@ -76,14 +82,31 @@ private void CheckWin()
...
-    public void GameOver(int index)
+    IEnumerator StopRestart()
     {
-    }
+        yield return new WaitForSeconds(1f);
+        SceneManager.LoadScene(SceneManager.GetActiveScene().buildIndex);
+        restartPanel.SetActive(false);
+        // Time.timeScale = 1;
+        enabled = true;
+    }*/
```
Comment (Eclmist) [inline Assets/Scripts/Gameplay/GameManager.cs]: Let's not push commented out code if possible

File: Assets/Scripts/Gameplay/Items/Item.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        
+        if (skillTransform == null)
+        {
+            Debug.LogError("Player should have a child GameObject named Skills");
+            return;
+        }
+
+        ItemSkill itemSkill = skillTransform.gameObject.AddComponent(m_ItemSkill.GetType()) as ItemSkill; //adds child type
+        itemSkill.InitializeSkill();
+        player.GetComponentInChildren<SkillHandler>().AddSkill(itemSkill); //must be after or else will be deleted
```
Comment (Eclmist) [inline Assets/Scripts/Gameplay/Items/Item.cs]: You should do a null check for the component before using it

File: Assets/Prefabs/Player/Player.prefab (hunk 1 above)
Comment (Eclmist) [reply Assets/Prefabs/Player/Player.prefab]: I'm also in the middle of some significant changes to the player prefab so don't push this file. Check with me in person to sort the data out

File: Assets/Scripts/Gameplay/Items/Item.cs (hunk 6 above)
Comment (tcgj) [reply Assets/Scripts/Gameplay/Items/Item.cs]: I suppose we should just have a private skill handler field within player for this then. Then make a getter. Will guarantee existence in this case.

Comment (Eclmist) [reply Assets/Scripts/Gameplay/Items/Item.cs]: Yeah but idk if the object is moved dynamically during runtime or not. Of course, if we can assign reference in the editor then do that instead 100%

File: Assets/Prefabs/Player/Player.prefab (hunk 1 above)
Comment (heze8) [reply Assets/Prefabs/Player/Player.prefab]: Okay lemme discuss with you on this

File: Assets/Scripts/AI/AiPlayer.cs (hunk 2 above)
Comment (heze8) [reply Assets/Scripts/AI/AiPlayer.cs]: Okay will remove until completed

File: Assets/Scripts/Gameplay/GameManager.cs (hunk 3 above)
Comment (heze8) [reply Assets/Scripts/Gameplay/GameManager.cs]: Alright I will remove it first

File: Assets/Scripts/AI/States/AiState.cs (hunk 4 above)
Comment (heze8) [reply Assets/Scripts/AI/States/AiState.cs]: Yes!

**Pull request closed without merging**